mono/metadata/class-init.c

```diff
@@ -5594,18 +5591,14 @@ MonoClass *
 mono_class_create_array_fill_type (void)
 {
 	static MonoClass klass;
-	static gboolean inited = FALSE;
 
-	if (!inited) {
-		klass.element_class = mono_defaults.int64_class;
-		klass.rank = 1;
-		klass.instance_size = MONO_SIZEOF_MONO_ARRAY;
-		klass.sizes.element_size = 8;
-		klass.size_inited = 1;
-		klass.name = "array_filler_type";
+	klass.element_class = mono_defaults.int64_class;
```

Comment:
This pattern does bounce cache lines.
Alternative is to read/check/write each field, so that cache lines remain shared, not bouncing exclusive.
In this case the caller is doing on demand initialization also so it is ok.

Reply:
C99 designated initializers seeming nice then. Only one of these is dynamic, and that could be addresses, perhaps, by adding an indirect bit.